libavformat/dashdec.c

```diff
@@ -1725,18 +1782,27 @@ static int get_current_fragment(struct representation *pls, struct fragment** ne
     if (c->is_live) {
         min_seq_no = calc_min_seg_no(pls->parent, pls);
         max_seq_no = calc_max_seg_no(pls, c);
+        old_max_seq_no = max_seq_no;
 
         if (pls->timelines || pls->fragments) {
             err = refresh_manifest(pls->parent);
             if (AVERROR_INPUT_CHANGED == err) {
                 return err;
             }
+            min_seq_no = calc_min_seg_no(pls->parent, pls);
+            max_seq_no = calc_max_seg_no(pls, c);
         }
         if (pls->cur_seq_no < min_seq_no) {
-            av_log(pls->parent, AV_LOG_VERBOSE, "old fragment: cur[%"PRId64"] min[%"PRId64"] max[%"PRId64"]\n", (int64_t)pls->cur_seq_no, min_seq_no, max_seq_no);
+            av_log(pls->parent, AV_LOG_VERBOSE, "old fragment: cur[%"PRId64"] min[%"PRId64"] max[%"PRId64"]\n",
+                   (int64_t)pls->cur_seq_no, min_seq_no, max_seq_no);
             pls->cur_seq_no = calc_cur_seg_no(pls->parent, pls);
```

Comment:
Would this skip forward if it fell too far behind?

Reply:
Looking at this more I am a bit worried, not about this code but about what happens when we do fall far behind. The fix_start_number function will just do nothing (effectively resetting the segment numbers)... but then shortly after it runs
```
ccur_video->cur_seq_no = FFMAX(newSeqNo, cur_video->cur_seq_no);
```

So seq_no will be outside the playlist and we will not find it (probably not till after we timeout/retry a bunch).

I think fix_start_number should either
- A) completely die if it falls too far behind
- B) slip ahead but keep numbering segments consecutively.. even if there is a discontinuity/gap

This particular code is fine and shouldn't be hit for timeline/segment based codes (since refresh manifest is right before it).

Reply:
OK, now when we startup or fall to far behind it will attempt to best-guess the segment using the timeline information